### ISSUE TITLE
Plugins!

### ISSUE THREAD

**Issue by ferndot** [codexa/firetext#129]
We need to structure Firetext's code so that the user can download parsers as plugins.  Also, the same thing could apply to cloud storage providers.

**Comment by HR** [comment]
plugins for what?

**Comment by ferndot** [comment]
Basically, we would provide an online "store" for plugins.  The user could choose the formats that he/she wants Firetext to be able to edit.

**Comment by ferndot** [comment]
E.G. Developers I have talked to want Markdown support.  Regular users wouldn't...

**Comment by HR** [comment]
ok, the we should have a developer mode and parsers for normal document formats(i.e. .docx .odt) should be supplied anyway by default. We want to make it as convenient as possible for our users

**Comment by ferndot** [comment]
Of course, we would just give advanced parsers in the store.  Maybe we could get 3rd parties involved so that there is a possibility for commercial formats.
